### PR TITLE
Refactor scheduler tests

### PR DESCRIPTION
**Pull request by mconiglio**
Refactored the following tests:

- job_presenter_spec
- manager_spec
- rufus_utils_spec
- schedule_spec
- utils_spec
- web_spec

### CODE REVIEW

File: spec/sidekiq-scheduler/job_presenter_spec.rb [hunk 1]
```diff
@@ -1,116 +1,99 @@
...
-      end
+      it { is_expected.to eq(job_presenter.relative_time(next_time)) }
     end
   end
 
-  describe "#interval" do
-    context "when the attributes has a cron key" do
-      let(:attributes) { { "cron" => cron_value } }
-      let(:cron_value) { "cron_value" }
+  describe '#interval' do
```
Comment (snmgian) [inline spec/sidekiq-scheduler/job_presenter_spec.rb]: There's no base case for these set of tests; either interval, nor every are edge cases.

So, in this situation, it would be ok to have smth like:

``` ruby
context 'with "cron" key' do
  // ...
end

context 'with "interval" key' do
  // ...
end

context 'with "every" key' do
  // ...
end
```

File: spec/sidekiq-scheduler/manager_spec.rb [hunk 2]
```diff
@@ -19,48 +16,25 @@
...
+      let(:enabled) { true }
 
-    it 'sets Sidekiq.schedule' do
-      expect {
-        subject
-      }.to change { Sidekiq.schedule }.to(options[:schedule])
+      it { expect { subject }.to change { Sidekiq::Scheduler.enabled }.to(options[:enabled]) }
+      it { expect { subject }.to change { Sidekiq::Scheduler.dynamic }.to(options[:dynamic]) }
+      it { expect { subject }.to change { Sidekiq::Scheduler.listened_queues_only }.to(options[:listened_queues_only]) }
+      it { expect { subject }.to change { Sidekiq.schedule }.to(options[:schedule]) }
```
Comment (snmgian) [inline spec/sidekiq-scheduler/manager_spec.rb]: Maybe this reads better ...
``` ruby
it { 
  expect { subject }.to change { Sidekiq.schedule }.to(options[:schedule]) 
}
```

File: spec/sidekiq-scheduler/schedule_spec.rb [hunk 3]
```diff
@@ -1,85 +1,60 @@
...
+      subject
 
-      expect(cron_hash).to eq(job_from_redis(job_id))
-    end
+      job = Store.job_from_redis(job_id)
 
-    it 'uses job name as \'class\' argument if it\'s missing' do
-      Sidekiq.schedule = {job_class_id => cron_hash.select(&only_cron_and_args)}
+      expect(job).to eq(schedule)
+      expect(job['class']).to eq(schedule['class'])
```
Comment (snmgian) [inline spec/sidekiq-scheduler/schedule_spec.rb]: If the previous expect passes, this will pass also. Remove it

File: spec/sidekiq-scheduler/schedule_spec.rb [hunk 4]
```diff
@@ -1,85 +1,60 @@
...
 
-      context 'when schedule is set twice' do
-        it 'sets the schedule on redis' do
-          2.times { Sidekiq.schedule = { job_id => cron_hash } }
+        job = Store.job_from_redis(job_id)
 
-          expect(cron_hash).to eq(job_from_redis(job_id))
-        end
+        expect(job).to eq(schedule.merge('class' => job_id))
+        expect(job['class']).to eq(job_id)
```
Comment (snmgian) [inline spec/sidekiq-scheduler/schedule_spec.rb]: idem

File: spec/sidekiq-scheduler/schedule_spec.rb [hunk 5]
```diff
@@ -1,85 +1,60 @@
...
+      job = Store.job_from_redis(job_id)
 
-    it 'uses job name as \'class\' argument if it\'s missing' do
-      Sidekiq.schedule = {job_class_id => cron_hash.select(&only_cron_and_args)}
+      expect(job).to eq(schedule)
+      expect(job['class']).to eq(schedule['class'])
+    end
 
-      job = Sidekiq.schedule[job_class_id]
+    context "when 'class' argument is not set" do
```
Comment (snmgian) [inline spec/sidekiq-scheduler/schedule_spec.rb]: `'when "class" argument is not set'`

File: spec/support/store.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+class Store
+  class << self
```
Comment (snmgian) [inline spec/support/store.rb]: use `def self.method` instead of `class << self`

File: spec/support/store.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
+class Store
```
Comment (snmgian) [inline spec/support/store.rb]: `module` instead of `class`

`Store` is a really generic name, some other gem could export a class/module with the same name and in that case we'll face some problems here

Comment (snmgian) [inline spec/support/store.rb]: scope it under SidekiqScheduler

File: spec/sidekiq-scheduler/job_presenter_spec.rb [hunk 8]
```diff
@@ -1,116 +1,99 @@
...
 
-  describe "#next_time" do
-    context "when the job has not a next time in redis" do
-      it "returns nothing" do
-        expect(subject.next_time).not_to be
-      end
+  describe '#next_time' do
+    subject { job_presenter.next_time }
+
+    context "when the job doesn't have a next time in redis" do
```
Comment (mconiglio) [inline spec/sidekiq-scheduler/job_presenter_spec.rb]: So you are saying that there shouldn't be a context here?

Comment (snmgian) [inline spec/sidekiq-scheduler/job_presenter_spec.rb]: That's an option. Another option is to set the appropriate context using a `before` block
